fahrschein/src/main/java/org/zalando/fahrschein/NakadiClient.java

```diff
@@ -95,8 +109,13 @@ public <T> void publish(String eventName, List<T> events) throws EventPublishing
             objectMapper.writeValue(body, events);
         }
 
+        eventPublishingHandlers.forEach(requestHandler -> requestHandler.onPublish(eventName, events));
```

Comment:
should we enclose this line within a `try-catch` block to guarantee that  `onError` and `afterPublish` gets executed. ?